### PR TITLE
Untouched relations get lost when contained in localizes fields in object bricks on save

### LINKED ISSUE #7302: Untouched relations get lost when contained in localizes fields in object bricks on save
# Bug Report
If you have a relation field (e.g. Many-To-One) in an localized field in an object brick and you are editing the content of this relation in one locale while leaving the other locales untouched, the content of the other locales is lost on saving the object.

### Expected behavior
If the content for one locale of a relation field in a localized object brick is changed, the content of the other locales should not get lost, when untouched.

### Actual behavior
The content of the other locales is lost.

### Steps to reproduce
Create an object brick with localized fields and a relation attribute (e.g. Many-To-One). Add this object brick to a class. Create an object of that class. Add the object brick to that object and fill the relation attribute for one locale. Save the object. Now reopen the object and fill the relation attribute for a second locale. Save again. After reloading the object, the content of the first locale is gone.

## Possibly related:
https://github.com/pimcore/pimcore/issues/7059 But the commit that solves the issue there, does not resolve the bug as described here.

### PR DESCRIPTION
**Pull request by weisswurstkanone**
resolves #7302

### CODE REVIEW

File: models/DataObject/Localizedfield/Dao.php [hunk 1]
```diff
@@ -135,9 +135,12 @@ public function save($params = [])
...
         foreach ($validLanguages as $language) {
-            if ((!isset($params['newParent']) || !$params['newParent']) && isset($params['isUpdate']) && $params['isUpdate'] && !$this->model->isLanguageDirty(
-                    $language
-                )) {
+            if ((!isset($params['newParent']) || !$params['newParent'])
+                && isset($params['isUpdate'])
+                && $params['isUpdate']
+                && !$this->model->isLanguageDirty($language)
+                 && !$forceUpdate
+                ) {
```
Comment (jdreesen) [inline models/DataObject/Localizedfield/Dao.php:143]: `!isset($params[...]) || !$params[...]` can be simplified to `empty($params[...])` 
and
`isset($params[...]) && $params[...]` can be simplified to `!empty($params[...])` 

```suggestion
            if (empty($params['newParent'])
                && !empty($params['isUpdate'])
                && !$this->model->isLanguageDirty($language)
                && !$forceUpdate
            ) {
```

**Comment by dold** [comment]
@weisswurstkanone  I just tried to patch our pimcore version (6.6.10) with the fix in models/DataObject/Localizedfield/Dao.php, but it still does not work. Can you give me any hint, if other parts of the code need to be updated? I can see, that there is no "isUpdate" attribute in the params array. And the $forceUpdate variable is not defined at all in the save method.
 
We have this as a quite pressing issue with a customer, but we cannot update the pimcore version right now to the latest release.

**Comment by weisswurstkanone** [comment]
No? https://github.com/pimcore/pimcore/blob/41c6a9106110449816aae6fcbfbdfaea1cbf0b7f/models/DataObject/Localizedfield/Dao.php#L101

**Comment by weisswurstkanone** [comment]
isUpdate set flag is set here
https://github.com/pimcore/pimcore/blob/4304973c91f2587be57ad683312e4cf82827a0c2/models/DataObject/Objectbrick/Data/Dao.php#L125

but this was there already before 6.6.10

**Comment by dold** [comment]
The $forceUpdate variable is not in 6.6.10. Let me check the isUpdate param.

**Comment by weisswurstkanone** [comment]
the forceUpdate flag was added 2 weeks ago. see https://github.com/pimcore/pimcore/commit/ebf150676da3e4da96866ad3bee1b5229dee4e43

the isUpdate param was there before

**Comment by dold** [comment]
Thanks. I guess there are then other files that also need to change, so that this is working. We are planning now an update to 6.8.2 and see if we can patch it there.

File: tests/Model/LazyLoading/ManyToManyRelationTest.php [hunk 2]
```diff
@@ -342,15 +342,33 @@ public function testLocalizedBrickAttributes()
...
+        $brick->getLocalizedfields()->setLocalizedValue('lrelations', $this->loadRelations()->load(), 'de');
 
         $object->getBricks()->setLazyLoadingLocalizedTest($brick);
         $object->save();
 
         $brick->setLInput(uniqid());
         $object->save();
         $object = Concrete::getById($object->getId(), true);
-        $this->assertTrue(count($object->getBricks()->getLazyLoadingLocalizedTest()->getLRelations()) > 0);
+        $this->assertTrue(count($object->getBricks()->getLazyLoadingLocalizedTest()->getLRelations('en')) > 0);
```
Comment (dvesh3) [inline tests/Model/LazyLoading/ManyToManyRelationTest.php:354]: test for other relation types?..at-least for many-to-one?

Comment (weisswurstkanone) [reply tests/Model/LazyLoading/ManyToManyRelationTest.php:362]: @dvesh3 added a test for many to one

**Review by dvesh3 — APPROVED** [review]

**Comment by Yivan** [comment]
@dold We get the problem to backport manually a fix on the same subject (see : https://github.com/pimcore/pimcore/issues/7192#issuecomment-702227966 ) it is linked I Think (fix has been availbale only starting from v6.7.3).
But we didn't manage to backport it... : ( So we have no choice to update Pimcore.
If you have any success I would be glad you give some hints on it !
Best regards.

**Pull request merged**